api/src/test/java/org/openmrs/CohortMembershipTest.java

```diff
@@ -252,4 +252,147 @@ public void compareTo_shouldSort() throws Exception {
+		
+		// Creating two logical identical CohortMemberships ot test equal
+		CohortMembership cohortMembershipOne = new CohortMembership(13);
+		CohortMembership cohortMembershipTwo = new CohortMembership(12);
+		
+		Date startDate1 = new Date(1545140935);
+		Date endDate1 = new Date(1577836801);
+		
+		Date startDate2 = new Date(1545140935);
+		Date endDate2 = new Date(1577836800);
+		
+		
+		cohortMembershipOne.setEndDate(endDate1);
+		cohortMembershipOne.setStartDate(startDate1);
+		cohortMembershipTwo.setStartDate(startDate2);
+		cohortMembershipTwo.setEndDate(endDate2);
+		
+		assertFalse(cohortMembershipOne.equals(cohortMembershipTwo));
+	}
+	
+	@Test
+	public void equal_shouldReturnFalseIfOnlyStartDateOfOneObjectIsNull() {
+		
+		CohortMembership cohortMembershipOne = new CohortMembership(13);
+		CohortMembership cohortMembershipTwo = new CohortMembership(12);
+		
+		Date startDate1 = null;
+		Date endDate1 = new Date(1577836801);
+		
+		Date startDate2 = new Date(1545140935);
+		Date endDate2 = null;
+		
+		
+		cohortMembershipOne.setEndDate(endDate1);
+		cohortMembershipOne.setStartDate(startDate1);
+		cohortMembershipTwo.setStartDate(startDate2);
+		cohortMembershipTwo.setEndDate(endDate2);
+		
+		assertFalse(cohortMembershipOne.equals(cohortMembershipTwo));
+	}
+	
+	@Test
+	public void equal_shouldReturnFalseIfOnlyEndDateOfOneObjectIsNull() {
+		
+		CohortMembership cohortMembershipOne = new CohortMembership(13);
+		CohortMembership cohortMembershipTwo = new CohortMembership(12);
+		long timestmap = 1577836801;
+		
+		Date startDate1 = new Date(timestmap);
+		Date endDate1 = null;
+		
+		Date startDate2 = new Date(timestmap);
+		Date endDate2 = null;
+		
+		
+		cohortMembershipOne.setEndDate(endDate1);
+		cohortMembershipOne.setStartDate(startDate1);
+		cohortMembershipTwo.setStartDate(startDate2);
+		cohortMembershipTwo.setEndDate(endDate2);
+		
+		assertFalse(cohortMembershipOne.equals(cohortMembershipTwo));
+	}
+	
+	@Test
+	public void equal_shouldReturnTrueIfPatientIdIsEqualAndStarDateAndEndDateAreBothNull() {
+		
+		// Creating two logical identical CohortMemberships to test equal method
+		CohortMembership cohortMembershipOne = new CohortMembership(12);
+		CohortMembership cohortMembershipTwo = new CohortMembership(12);
+		
+		Date startDate = null;
+		Date endDate = null;
+		
+		cohortMembershipOne.setEndDate(endDate);
+		cohortMembershipOne.setStartDate(startDate);
+		cohortMembershipTwo.setStartDate(startDate);
+		cohortMembershipTwo.setEndDate(endDate);
+		
+		assertTrue(cohortMembershipOne.equals(cohortMembershipTwo));
+		
+	}
+	
+	@Test
+	public void equal_shouldReturnTrueWhenBothPatientIdsAreNullAndEndDateAndStartDateAreEqual() {
+		
+		// Creating two logical identical CohortMemberships to test equal method
+		CohortMembership cohortMembershipOne = new CohortMembership();
+		CohortMembership cohortMembershipTwo = new CohortMembership();
+		
+		Date startDate1 = new Date(1545140935);
```

Comment:
Do wee need to duplicate 1545140935?

Reply:
I will make a variable out of it. And yeah I think it is helpful for testing purposes to duplicate it

Reply:
how would duplicating it be helpful?

Reply:
The alternative would be to say: startDate2 = startDate1 correct?
But if I am doing so the objects point to the same area and the reference is already equal. I want to test/make sure that the equal() function of the Date type is executed. Hence to verify, if both dates have the same content. That is why I need to duplicate the value to create a new object which is equal but not the same

Reply:
Assigning that value to a constant always helps to avoid the duplication 😊

Reply:
I always used a constant instead of duplicating the same value twice. But I am wondering: What is bad about that duplication? 

Because noone sees that these values were supposed to be the same? Does it has to be the same for the patientId in the constructors?

Reply:
If you changed the value of one, would you change the other too?

Reply:
What do you mean? I can't follow your right now :(

Reply:
If you wanted to change that value from 1545140935 to 1545140936, would also change it for the second date?

Reply:
Yes. it should change the value of both startDate's. The one of object1 and object2. Because I want to test if it fails, when only the endDate is different in both objects